infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -154,39 +162,80 @@ private void checkLimits() {
         }
     }
 
+    /**
+     * This class provides resources required for PartitionAttribute and its usages.
+     * PartitionAttribute has attributes related to the partition containing the log files like
+     * readOnly, availableSpace and totalSpace that are refreshed every
+     * {@link PartitionAttribute#UPDATE_INTERVAL} seconds using the
+     * {@link PartitionAttribute#scheduler}.
+     */
     public static class PartitionAgent {
+
+        // Interval when the PartitionAttribute values are reset by the scheduler
+        private static final int UPDATE_INTERVAL = 5;
+
+        // We don't need any delay for the scheduler
+        private static final int NO_DELAY = 0;
+
+        // This contains the attribute values of the log partition
         @Getter
         private PartitionAttribute partitionAttribute;
```

Comment:
Does this need to be marked as volatile since the value is mutated from a separate thread within the scheduler?

Reply:
Good catch, Done, Thank you.